File: reset-samples/python/reset-point/activities.py
"""Activities for the order workflow."""
import asyncio
from temporalio import activity


@activity.defn
async def validate_order(order_id: str) -> str:
    """Validate an order."""
    activity.logger.info(f"Validating order {order_id}")
    # Simulate validation work
    await asyncio.sleep(0.1)
    return f"Validated-{order_id}"


@activity.defn
async def process_payment(order_id: str) -> str:
    """Process payment for an order."""
    activity.logger.info(f"Processing payment for {order_id}")
    # Simulate payment processing
    await asyncio.sleep(0.1)
    return f"Paid-{order_id}"


@activity.defn
async def ship_order(order_id: str) -> str:
    """Ship an order."""
    activity.logger.info(f"Shipping order {order_id}")
    # Simulate shipping
    await asyncio.sleep(0.1)
    return f"Shipped-{order_id}"

File: reset-samples/typescript/reset-point/src/activities.ts
/**
 * Activities for the order workflow.
 */

/**
 * Validate an order.
 */
export async function validateOrder(orderId: string): Promise<string> {
  console.log(`Validating order ${orderId}`);
  // Simulate validation work
  await new Promise((resolve) => setTimeout(resolve, 100));
  return `Validated-${orderId}`;
}

/**
 * Process payment for an order.
 */
export async function processPayment(orderId: string): Promise<string> {
  console.log(`Processing payment for ${orderId}`);
  // Simulate payment processing
  await new Promise((resolve) => setTimeout(resolve, 100));
  return `Paid-${orderId}`;
}

/**
 * Ship an order.
 */
export async function shipOrder(orderId: string): Promise<string> {
  console.log(`Shipping order ${orderId}`);
  // Simulate shipping
  await new Promise((resolve) => setTimeout(resolve, 100));
  return `Shipped-${orderId}`;
}

